extensions/amp-image-slider/0.1/test/integration/test-amp-image-slider.js

```diff
@@ -446,5 +470,95 @@ config.run('amp-image-slider', function() {
             return pollByBarLeftPos(rect.right);
           });
     });
+
+    it('should seekTo correct position', () => {
+      let button;
+      let pos10Percent;
+      return prep()
+          .then(() => {
+            button = doc.querySelector('#button');
+            pos10Percent = rect.left + 0.1 * rect.width;
+            button.click();
+            return pollByBarLeftPos(pos10Percent);
+          });
+    });
+
+    it('should show hint again on scroll back and into viewport', () => {
+      let slider, container_, hint;
+      let mouseDownEvent;
+      return prep()
+          .then(() => {
+            slider = doc.querySelector('#slider');
+            container_ = slider
+                .querySelector('.i-amphtml-image-slider-container');
+            hint = slider.querySelectorAll('.i-amphtml-image-slider-hint')[0];
+            mouseDownEvent = createFakeEvent(
+                'mousedown',
+                0,
+                0
+            );
+            container_.dispatchEvent(mouseDownEvent);
+            return poll('should receive hidden class', () => {
+              return hint.classList
+                  .contains('i-amphtml-image-slider-hint-hidden');
+            }, () => {});
+          })
+          .then(() => {
+            env.win.scrollTo({
+              top: 3000,
+            });
+            // Have to use timeout here,
+            // no indication of proper viewportCallback trigger
+            return timeout(env.win, 500);
```

Comment:
This part isn't clear to me, is the point of this so that the next scrollTo causes viewport callback to trigger?

There should be a better way of doing this (e.g. overriding viewportCallback and resolving a Promise, then calling through to the original viewportCallback).

Alternatively, just do a poll for `isInViewport()` to become false for `slider.implementation_`.

Reply:
I could not override any methods or properties for integration tests (with --pseudo_names).

I can check if `isInViewport()` is exposed, but I believe `--pseudo_names` probably also masked this. On the other hand, I would also need to double check if `isInViewport()` would always reflect whether `viewportCallback` is triggered already, or forcing the latter to trigger before returning from the call